### PR TITLE
deployment flow doesn't work on MAAS

### LINKED ISSUE #2361: unable to work with models on maas, critical error
Juju GUI 2.2.7 on Juju 2.1beta4 on guimaas.

Loading the gui and getting the controller model (we should default to default as well?)

    combo?app/assets/javascripts/yui/lazy-model-list/lazy-model-list-min.js&app/assets/javascripts/yui/…:36 Uncaught TypeError: Cannot read property 'value' of undefined
        at f.maas.h.provider.modelGet.y (combo?app/assets/javascripts/yui/lazy-model-list/lazy-model-list-min.js&app/assets/javascripts/yui/…:36)
        at r.<anonymous> (combo?app/assets/javascripts/yui/lazy-model-list/lazy-model-list-min.js&app/assets/javascripts/yui/…:36)
        at r.dispatch_result (combo?app/assets/javascripts/yui/lazy-model-list/lazy-model-list-min.js&app/assets/javascripts/yui/…:36)
        at e.CustomEvent.u.fireComplex (combo?app/assets/javascripts/yui/oop/oop-min.js&app/assets/javascripts/yui/event-custom-base/event-…:9)
        at e.CustomEvent._fire (combo?app/assets/javascripts/yui/oop/oop-min.js&app/assets/javascripts/yui/event-custom-base/event-…:4)
        at r.fire (combo?app/assets/javascripts/yui/oop/oop-min.js&app/assets/javascripts/yui/event-custom-base/event-…:5)
        at r.on_message (combo?app/assets/javascripts/yui/widget-buttons/widget-buttons-min.js&app/assets/javascripts/yui/wi…:42)
        at WebSocket.h.onmessage (combo?app/store/env/bakery-min.js&app/jujulib/index-min.js&app/jujulib/charmstore-min.js&app/jujuli…:13)

In chatting with @hatched it seems that we're getting/not getting expected data from the maas provider connection in api.js. Line #817 in api.js was mentioned but that might be skewed based on what branch/code @hatched was looking at. 

After this error we're unable to go to the profile page and unable to create a new model at all. These error seems like the first step in validating the add-model will work as well. However, there might be more than one bug there.

### PR DESCRIPTION
**Pull request by mitechie**
Fixes #2361

- MAAS does not have any regions
- Update _parseCloudResults to handle that properly
- Add a test to the controller API tests that verifies that works

Drive-by
- Make the karma.conf.js more specific so that it runs tests properly on
Rick's system.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6635/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6636/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6653/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6654/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6655/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6656/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6657/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6658/

**Comment by mitechie** [comment]
Perfomed a manual QA of this on guimaas. The Juju GUI running there is a make dist of this PR. On it I've successfully created a new model and deployed a charm through the deployment flow successfully.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6659/

**Comment by mitechie** [comment]
Yay, made the linter happy to get a passing build

File: karma.conf.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ module.exports = function(config) {
     // set the options for the various preprocessors used
     babelPreprocessor: {
       filename: function (file) {
-        return file.originalPath.replace(/\/src\//, '/build/');
+        return file.originalPath.replace(/gui\/src\//, 'gui/build/');
```
Comment (hatched) [inline karma.conf.js]: Thanks for this improvement :)

File: jujugui/static/gui/src/test/test_controller_api.js [hunk 2]
```diff
@@ -1352,7 +1352,7 @@ describe('Controller API', function() {
       // Perform the request.
       controllerAPI.listClouds((err, clouds) => {
         assert.strictEqual(err, null);
-        assert.deepEqual(clouds, {
+        let expected = {
```
Comment (mitechie) [inline jujugui/static/gui/src/test/test_controller_api.js]: no reason, updated.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6660/

**Comment by mitechie** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**